apps/emqx/test/emqx_authentication_SUITE.erl

```diff
@@ -326,14 +286,14 @@ t_update_config(Config) when is_list(Config) ->
     ok = register_provider(?config("auth2"), ?MODULE),
     Global = ?config(global),
     AuthenticatorConfig1 = #{
-        <<"mechanism">> => <<"password_based">>,
-        <<"backend">> => <<"built_in_database">>,
-        <<"enable">> => true
+        mechanism => password_based,
+        backend => built_in_database,
+        enable => true
```

Comment:
Why do we need to change to atom? 🤔

Reply:
It's because this SUITE is now testing configs without type check.
Previously, there was a (hidden API contract) `check_config/1` callback (implemented in this module, see old code) which does the type check.
--- Having to support this callback type check was essentially the reason why we had to use LAZY type and caused all the complexity!

Now, we have to admit that this way of extending authn functionality is over-engineered:
it's way much more straightforward to just add another Erlang app (e.g. a plugin) which hooks to 'client.authenticate' hook point.